src/examples/mfem_mortar_lm_patch.cpp

```diff
@@ -0,0 +1,293 @@
+    mesh.reset(nullptr);
+
+    // further refinement of parallel mesh
+    {
+      int par_ref_levels = 0;
+      for (int i{0}; i < par_ref_levels; ++i)
+      {
+        pmesh->UniformRefinement();
+      }
+    }
+  }
+  timer.stop();
+  SLIC_INFO_ROOT(axom::fmt::format(
+    "Time to create parallel mesh: {0:f}ms", timer.elapsedTimeInMilliSec()
+  ));
+  
+  // set up data collection for output
+  auto pv_dc = mfem::ParaViewDataCollection("mortar_patch_pv", pmesh.get());
+  auto vi_dc = mfem::VisItDataCollection("mortar_patch_vi", pmesh.get());
+
+  // grid function for higher-order nodes
+  timer.start();
+  auto fe_coll = mfem::H1_FECollection(order, pmesh->SpaceDimension());
+  auto par_fe_space = mfem::ParFiniteElementSpace(
+    pmesh.get(), &fe_coll, pmesh->SpaceDimension());
+  auto coords = mfem::ParGridFunction(&par_fe_space);
+  if (order > 1)
+  {
+    pmesh->SetNodalGridFunction(&coords, false);
+  }
+  else
+  {
+    pmesh->GetNodes(coords);
+  }
+  pv_dc.RegisterField("pos", &coords);
+  vi_dc.RegisterField("pos", &coords);
+
+  // grid function for displacement
+  mfem::ParGridFunction u { &par_fe_space };
+  pv_dc.RegisterField("disp", &u);
+  vi_dc.RegisterField("disp", &u);
+  u = 0.0;
+  timer.stop();
+  SLIC_INFO_ROOT(axom::fmt::format(
+    "Time to create grid functions: {0:f}ms", timer.elapsedTimeInMilliSec()
+  ));
+
+  // save initial configuration
+  pv_dc.Save();
+  vi_dc.Save();
+
+  // recover dirichlet bc tdof list
+  timer.start();
+  mfem::Array<int> ess_tdof_list;
+  {
+    mfem::Array<int> ess_vdof_marker;
+    mfem::Array<int> ess_bdr(pmesh->bdr_attributes.Max());
+    ess_bdr = 0;
+    for (auto xfix_attrib : xfix_attribs)
+    {
+      ess_bdr[xfix_attrib-1] = 1;
+    }
+    par_fe_space.GetEssentialVDofs(ess_bdr, ess_vdof_marker, 0);
+    mfem::Array<int> new_ess_vdof_marker;
+    ess_bdr = 0;
+    for (auto yfix_attrib : yfix_attribs)
+    {
+      ess_bdr[yfix_attrib-1] = 1;
+    }
+    par_fe_space.GetEssentialVDofs(ess_bdr, new_ess_vdof_marker, 1);
+    for (int i{0}; i < ess_vdof_marker.Size(); ++i)
+    {
+      ess_vdof_marker[i] = ess_vdof_marker[i] || new_ess_vdof_marker[i];
+    }
+    ess_bdr = 0;
+    for (auto zfix_attrib : zfix_attribs)
+    {
+      ess_bdr[zfix_attrib-1] = 1;
+    }
+    par_fe_space.GetEssentialVDofs(ess_bdr, new_ess_vdof_marker, 2);
+    for (int i{0}; i < ess_vdof_marker.Size(); ++i)
+    {
+      ess_vdof_marker[i] = ess_vdof_marker[i] || new_ess_vdof_marker[i];
+    }
+    mfem::Array<int> ess_tdof_marker;
+    par_fe_space.GetRestrictionMatrix()->BooleanMult(ess_vdof_marker, ess_tdof_marker);
+    mfem::FiniteElementSpace::MarkerToList(ess_tdof_marker, ess_tdof_list);
+  }
+  timer.stop();
+  SLIC_INFO_ROOT(axom::fmt::format(
+    "Time to set up boundary conditions: {0:f}ms", timer.elapsedTimeInMilliSec()
+  ));
+
+  // set up mfem elasticity bilinear form
+  timer.start();
+  mfem::ParBilinearForm a(&par_fe_space);
+  mfem::ConstantCoefficient lambda(50.0);
```

Comment:
Tune to actual material?

Reply:
I made them changeable parameters in the input.